src/clients/Device.js

```diff
@@ -203,27 +203,35 @@ class Device extends EventEmitter {
         decipherStream,
         deviceID,
         handshakeBuffer,
-        pendingBuffers,
       } = await this._handshake.start(this);
+
       this._id = deviceID;
+      this._cipherStream = cipherStream;
+      this._decipherStream = decipherStream;
 
       this._getHello(handshakeBuffer);
-      this._sendHello(cipherStream, decipherStream);
+    } catch (error) {
+      this.disconnect(error);
+      throw error;
+    }
+  };
 
-      pendingBuffers.map((data: Buffer): void => this.routeMessage(data));
-      decipherStream.on('readable', () => {
-        process.nextTick(() => {
-          const chunk = ((decipherStream.read(): any): Buffer);
+  completeProtocolInitialization = () => {
+    try {
+      this._sendHello();
+      nullthrows(this._decipherStream).on(
```

Comment:
consider: You can put `this._decipherStream` in a temporary variable so you don't have to do multiple nullthrows